File: Python/DateOfEaster.py
#!/usr/bin/env python

# Date of Easter
# Based on algorithm by Jean Meeus, Astronomical Algorithms, 1998, p. 67

while True:
    x = int(raw_input("Enter a year:"))
    if x >= 1583:
        break
    print("Year earlier than 1583 is not allowed!")

a = x % 19
b = x / 100
c = x % 100
d = b / 4
e = b % 4
f = (b + 8) / 25
g = (b - f + 1) / 3
h = (19 * a + b - d - g + 15) % 30
i = c / 4
k = c % 4
l = (32 + 2 * e + 2 * i - h - k) % 7
m = (a + 11 * h + 22 * l) / 451
z = h + l - 7 * m + 114
n = z / 31
p = z % 31

if n == 3:
    print("March " + str(p+ 1))
elif n == 4:
    print("April " + str(p + 1))

File: Lua/DateOfEaster.lua
#!/usr/bin/lua

-- Date of Easter
-- Based on algorithm by Jean Meeus, Astronomical Algorithms, 1998, p. 67

local x

repeat
    io.write('Enter a year: ')
    x = tonumber(io.read())
    if (x == nil) then
        io.write("Not a number!\n")
    elseif (x < 1583) then
        io.write("Year earlier than 1583 is not allowed!\n")
    else
        break
    end
until (false)

local a = x % 19
local b = x // 100
local c = x % 100
local d = b // 4
local e = b % 4
local f = (b + 8) // 25
local g = (b - f + 1) // 3
local h = (19 * a + b - d - g + 15) % 30
local i = c // 4
local k = c % 4
local l = (32 + 2 * e + 2 * i - h - k) % 7
local m = (a + 11 * h + 22 * l) // 451
local z = h + l - 7 * m + 114
local n = z // 31
local p = z % 31

if (n == 3) then
    io.write('March ', p + 1, '\n')
elseif (n == 4) then
    io.write('April ', p + 1, '\n')
end
